srk/src/transition.ml

```diff
@@ -136,8 +135,12 @@ struct
             | Some t -> t
             | None -> mk_const srk (Var.symbol_of v)
           in
-          left_eq := (mk_eq srk left_term phi)::(!left_eq);
-          right_eq := (mk_eq srk right_term phi)::(!right_eq);
+          left_eq := (match Term.refine srk left_term with 
```

Comment:
Here and a couple of other places we have the same logic, in which we *want* to equate two terms regardless of type but can't.  I suggest extracting this logic to a single function in syntax.ml.